CommonVCs/CategoryVC.swift

```diff
@@ -0,0 +1,75 @@
+//
+//  CategoryVC.swift
+//  To-Do App
+//
+//  Created by Ananth Desai on 18/02/22.
+//
+
+import UIKit
+
+class CategoryVC: UIViewController, UITableViewDelegate, UITableViewDataSource {
+    
+    var selectedIndex: Int = 0
+    
+    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return sourceData.count
+    }
+    
+    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = returnTableView.dequeueReusableCell(withIdentifier: "Cell", for: indexPath)
+        cell.textLabel?.text = sourceData[indexPath.row].title
+        cell.backgroundColor = .systemGreen
+        cell.accessoryType = .disclosureIndicator
+        return cell
+    }
+    
+    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        let newVC = ItemsVC(indexPath.row)
+        selectedIndex = indexPath.row
+        newVC.title = sourceData[indexPath.row].title
+        navigationController?.pushViewController(newVC, animated: true)
+    }
+    
+    let returnTableView: UITableView = {
+        let table = UITableView()
+        table.register(UITableViewCell.self, forCellReuseIdentifier: "Cell")
+        return table
+    }()
```

Comment:
You shouldn't define class properties where ever you like. It should be declared on top of class. Also delegate extensions should be in extension.